datafusion/expr/src/expr.rs

```diff
@@ -1475,15 +1476,19 @@ impl Expr {
     /// Used when the expression forms the output field of a certain plan.
     /// The result is the field's qualifier and field name in the plan's
     /// output schema. We can use this qualified name to reference the field.
-    pub fn qualified_name(&self) -> (Option<TableReference>, String) {
+    ///
+    /// Note returns a Cow for the name to avoid unnecessary allocations.
+    pub fn qualified_name(&self) -> (Option<TableReference>, Cow<'_, String>) {
```

Comment:
having to return a String here is super unfortunate as it requires cloneing all the time